### PR TITLE
PRIME-2066 Cancel Login Redirect with Alert

### PR DESCRIPTION
**Pull request by johnnyl6**
[PRIME-2066](https://bcgovmoh.atlassian.net/browse/PRIME-2066?atlOrigin=eyJpIjoiYjVmZjg5ZDViNGIwNDAxYWFkYzMxNjk4MTcyOTE3ZjQiLCJwIjoiaiJ9)

- added cancel login redirect with alert

Note: can only be tested manually for now by appending `action=cancelled` to the login url until BCSC implements the `action` query string.

### CODE REVIEW

**Review by AMiKaH — COMMENTED** [review]
Can't seem to be redirected or have a warning show up when I cancel.

**Comment by johnnyl6** [comment]
> > [PRIME-2066](https://bcgovmoh.atlassian.net/browse/PRIME-2066?atlOrigin=eyJpIjoiYjVmZjg5ZDViNGIwNDAxYWFkYzMxNjk4MTcyOTE3ZjQiLCJwIjoiaiJ9)
> > 
> > * added cancel login redirect with alert
> > 
> > Note: can only be tested manually for now until BCSC implements the `action` query string.
> 
> How can it be "tested manually"?

@neophyte57  @AMiKaH  append query param `action=cancelled` to the login page url, and the alert will appear.

**Comment by neophyte57** [comment]
> > > [PRIME-2066](https://bcgovmoh.atlassian.net/browse/PRIME-2066?atlOrigin=eyJpIjoiYjVmZjg5ZDViNGIwNDAxYWFkYzMxNjk4MTcyOTE3ZjQiLCJwIjoiaiJ9)
> > > 
> > > * added cancel login redirect with alert
> > > 
> > > Note: can only be tested manually for now until BCSC implements the `action` query string.
> > 
> > 
> > How can it be "tested manually"?
> 
> @neophyte57 @AMiKaH append query param `action=cancelled` to the login page url, and the alert will appear.

What would be involved to get BCSC team to update the URL that is currently going to https://develop.pharmanetenrolment.gov.bc.ca/info ?

image

**Review by AMiKaH — APPROVED** [review]

File: prime-angular-frontend/src/app/shared/components/auth/prime-enrolment-access/prime-enrolment-access.component.html [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
+        <p>You have cancelled your login to PRIME</p>
+        <p>For assistance with PRIME, contact {{ primeSupportPhoneNumber | phone }} or
+          <a [attr.href]="'mailto:' + primeSupportEmail">{{ primeSupportEmail }}</a>
+        <p>
+          For assistance logging in with the BC Services Card app, contact the Helpdesk
+          <a
+              target="_blank"
+              rel="noopener noreferrer"
+              [attr.href]="bcscHelpDeskUrl">
+              here</a>.
```
Comment (hogmann) [inline prime-angular-frontend/src/app/shared/components/auth/prime-enrolment-access/prime-enrolment-access.component.html:22]: This here and the other clickable links are hard to make out in this alert. Can we see about making it more visible?

File: prime-angular-frontend/src/app/shared/components/auth/prime-enrolment-access/prime-enrolment-access.component.html [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+               icon="error_outline">
+      <ng-container #alertTitle
+                    class="alert-title">
+        Login Cancelled
+      </ng-container>
+      <ng-container #alertContent
+                    class="alert-content">
+        <p>You have cancelled your login to PRIME</p>
+        <p>For assistance with PRIME, contact {{ primeSupportPhoneNumber | phone }} or
+          <a [attr.href]="'mailto:' + primeSupportEmail">{{ primeSupportEmail }}</a>
```
Comment (hogmann) [inline prime-angular-frontend/src/app/shared/components/auth/prime-enrolment-access/prime-enrolment-access.component.html:15]: So there is some mixed review but i thought we wanted PRIMESupport@gov.bc.ca, but i see on the bottom of the page it is not capitalized. So curious if this is the case.

**Review by hogmann — APPROVED** [review]
Everything works great, just two small questions but could be non-actions. you can close them if that is the case.

File: prime-angular-frontend/src/app/shared/components/auth/prime-enrolment-access/prime-enrolment-access.component.html (hunk 1 above)
Comment (johnnyl6) [reply prime-angular-frontend/src/app/shared/components/auth/prime-enrolment-access/prime-enrolment-access.component.html:22]: agreed. underlined `here` to be more visible as the mockup.

File: prime-angular-frontend/src/app/shared/components/auth/prime-enrolment-access/prime-enrolment-access.component.html (hunk 2 above)
Comment (johnnyl6) [reply prime-angular-frontend/src/app/shared/components/auth/prime-enrolment-access/prime-enrolment-access.component.html:15]: good point. now using the prime phone and prime email component to make them consistent throughout the app.

**Pull request merged**